### PR TITLE
Migrate to npm 7

### PR DESCRIPTION
**Pull request by skjnldsv**
Yet another shot at it.
Lots of outdated sections :weary: 

I didn't change the babel config as this is causing export issues and make everything not defined: `dav is not defined` `OC is not defined`  etc etc. Most likely an issue with umd/commonjs exports

___
Followups:
- [ ] Address the `vue/no-mutating-props`
  https://github.com/nextcloud/server/blob/7ca81f360f42341b2941798374160d362507ba26/.eslintrc.js#L18
- [ ] Migrate from jsunit to jest

### CODE REVIEW

**Comment by juliushaertl** [comment]
>  Migrate from jsunit to ....? shrug

We already have a jest setup for core/src/tests https://github.com/nextcloud/server/pull/28115/checks?check_run_id=3133484042

**Comment by skjnldsv** [comment]
NODB failure is unrelated

**Review by juliusknorr — APPROVED** [review]
:partying_face:

**Comment by szaimen** [comment]
Did you push the production or dev js?

**Comment by skjnldsv** [comment]
> Did you push the production or dev js?

image

File: core/src/components/login/LoginForm.vue [hunk 1]
```diff
@@ -140,7 +140,8 @@ export default {
 			default: '',
 		},
 		redirectUrl: {
-			type: String,
+			type: [String, Boolean],
+			default: false,
```
Comment (artonge) [inline core/src/components/login/LoginForm.vue:144]: Brrr, don't like mixed types. Can't `undefined` do the job ?

File: core/src/components/login/PasswordLessLoginForm.vue [hunk 2]
```diff
@@ -71,7 +71,8 @@ export default {
 			default: '',
 		},
 		redirectUrl: {
-			type: String,
+			type: [String, Boolean],
+			default: false,
```
Comment (artonge) [inline core/src/components/login/PasswordLessLoginForm.vue:75]: Brrr, don't like mixed types. Can't `undefined` do the job ?

**Comment by szaimen** [comment]
> > Did you push the production or dev js?
> 
> image

So this checks also if the correct js was pusehd?

File: core/src/views/Login.vue [hunk 3]
```diff
@@ -111,91 +111,61 @@
...
 import PasswordLessLoginForm from '../components/login/PasswordLessLoginForm.vue'
 import ResetPassword from '../components/login/ResetPassword.vue'
 import UpdatePassword from '../components/login/UpdatePassword.vue'
 
+const query = queryString.parse(location.search)
+if (query.clear === '1') {
+	try {
+		window.localStorage.clear()
+		window.sessionStorage.clear()
+		console.debug('Browser storage cleared')
```
Comment (artonge) [inline core/src/views/Login.vue:127]: No `@nextcloud/logger` ?

**Comment by skjnldsv** [comment]
> So this checks also if the correct js was pushed?

Yes, it always have been :wink: 
https://github.com/nextcloud/server/blob/717636fd2815bdd49599af059965a8202b1582aa/.github/workflows/node.yml#L44-L52

**Comment by szaimen** [comment]
okay and shouldn't we wait until the tests finish?

**Comment by skjnldsv** [comment]
> okay and shouldn't we wait until the tests finish?

Finished! :bellhop_bell: :grin:

**Review by szaimen — APPROVED** [review]

File: .eslintrc.js [hunk 4]
```diff
@@ -13,5 +13,8 @@ module.exports = {
 	extends: ['@nextcloud'],
 	rules: {
 		'no-tabs': 'warn',
-	}
+		// TODO: make sure we fix this as this is bad vue coding style.
+		// Use proper sync modifier
+		'vue/no-mutating-props': 'warn',
```
Comment (artonge) [inline .eslintrc.js:18]: Only three remaining:

- `/home/runner/work/server/server/apps/workflowengine/src/components/Check.vue:22`
- `/home/runner/work/server/server/apps/workflowengine/src/components/Check.vue:30`
- `/home/runner/work/server/server/apps/workflowengine/src/components/Rule.vue:30`

**Review by artonge — DISMISSED** [review]

**Comment by artonge** [comment]
> > > Did you push the production or dev js?
> > 
> > 
> > image
> 
> So this checks also if the correct js was pushed?

Yes. It compile in production mode, then check if nothing as changed

File: core/src/views/Login.vue (hunk 3 above)
Comment (skjnldsv) [reply core/src/views/Login.vue:127]: This is code move, not new code.

File: .eslintrc.js (hunk 4 above)
Comment (skjnldsv) [reply .eslintrc.js:18]: Not fixable as they're in the template
https://github.com/nextcloud/server/issues/28119

File: core/src/components/login/PasswordLessLoginForm.vue (hunk 2 above)
Comment (skjnldsv) [reply core/src/components/login/PasswordLessLoginForm.vue:75]: No, this si a bad coding from the loadState library.
You cannot have a fallback value as `undefined` or it still throw and says `initial state cannot be found`

https://github.com/nextcloud/nextcloud-initial-state/blob/cbf8a415e5ed63675eeed74da336e9150349077a/lib/index.ts#L10
https://github.com/nextcloud/server/blob/f14b8aa34bfb9f7af7b23cfecc09e2fb8f604c1b/core/src/login.js#L61

File: core/src/components/login/LoginForm.vue (hunk 1 above)
Comment (skjnldsv) [reply core/src/components/login/LoginForm.vue:144]: See https://github.com/nextcloud/server/pull/28115#discussion_r674821886

File: core/src/components/login/PasswordLessLoginForm.vue (hunk 2 above)
Comment (artonge) [reply core/src/components/login/PasswordLessLoginForm.vue:75]: Sad, but ok

**Pull request merged**